enhancements/cluster-logging/logs-based-alerts.md

```diff
@@ -0,0 +1,300 @@
+
+[AlertingRule][alertingrule-crd] CRD fullfills the requirement to define alerting rules for Loki similar to `PrometheusRule`.
+
+[RulerConfig][rulerconfig-crd] CRD fullfills the requirement to connect the Loki Ruler component to notify a list of Prometheus AlertManager hosts on firing alerts.
+
+### User Stories
+
+* As an application owner, I want to use AlertingRule custom resources to notify AlertManager on firing alerts from my workload logs.
+* As a cluster admin, I want to use AlertingRule custom resources to notify AlertManager on firing alerts from the platform logs.
+* As a cluster admin, I want to use AlertingRule custom resources to notify AlertManager on firing alerts from the audit logs.
+
+### Goals
+
+* Application Owners can configure alert notifications for application alerts based on workload logs.
+* Cluster admins can configure alert notifications for platform alerts based on platform logs.
+* Cluster admins can configure alert notifications for platform alerts based on audit logs.
+
+### Non-Goals
+
+* Specific alerting support on the OCP console for Logs-based alerts. This should be seamless in the existing alerting UIs.
+* Recording rules to compile metrics from logs.
+
+## Proposal
+
+We plan to leverage the `AlertingRule` and `RulerConfig` custom resource definitions already exposed by the Loki Operator so that:
+- cluster admins can configure how the platform logs can notify on alertworthy platform and/or audit logs.
+- application owners can configure how the workload logs can notify on alertworthy logs.
+
+### Workflow Description
+
+#### Logs-based Alerts for Application Owners
+
+1. The application owner creates an AlertingRule custom resource on a workload project, e.g.
+
+```yaml
+apiVersion: loki.grafana.com/v1beta1
+kind: AlertingRule
+metadata:
+  name: my-workload-alerts
+  namespace: my-workload-ns
+  labels:
+    openshift.io/cluster-monitoring: "true"
+spec:
+  tenantID: application
+  groups:
+    - name: MyApplication
+      rules:
+        - alert: MyApplicationHighPercentageError
+          expr: |
+            sum(rate({kubernetes_namespace_name="my-workload-ns", kubernetes_pod_name=~"my-workload.*"} |= "error" [1m])) by (job)
+              /
+            sum(rate({kubernetes_namespace_name="my-workload-ns", kuberentes_pod_name=~"my-workload.*"}[1m])) by (job)
+              > 100
+          for: 5m
+          labels:
+            severity: warning
+          annotations:
+            summary: My Application High Errors.
+            description: My application throws a high amount of error logs.
+```
+
+2. The application owner can only define AlertingRules custom resources in project with granted access to them and only for the tenant `application`.
```

Comment:
> only for the tenant `application`.

will this be enforced?

Reply:
I have not much experience with AdmissionWebhooks, but I would to think about this like:
- Application owners: `oc create -n my-workload-ns alertingrule my-workload-alerts` where `spec.tenantID: application` succeeds
- Only members of `cluster-admins` can create with `spec.tenantID: infrastructure` and/or `spec.tenantID: audit`.

WDYT?

Reply:
This seems feasible but it probably deserves input from the API folks. FWIW we run an admission webhook for PrometheusRules: this isn't trivial but still doable (think about proper HA for instance).